src/compiler/crystal/semantic/main_visitor.cr

```diff
@@ -2469,7 +2469,7 @@ module Crystal
       # (useful for sizeof inside as a generic type argument, but also
       # to make it easier for LLVM to optimize things)
       if !node.exp.is_a?(TypeOf)
-        expanded = NumberLiteral.new(@program.size_of(type.sizeof_type))
+        expanded = NumberLiteral.new(@program.size_of(type.sizeof_type).to_s, :i32)
```

Comment:
Maybe add a comment to explain why it's important to make them `:i32` here?

(and ditto below)

Reply:
I don't even really know, why exactly this is necessary. 😅 The type restriction can be removed when codegen bug is fixed.

Reply:
There's no codegen bug. It was working previously.

Reply:
If there's no bug, why are the second and third commit necessary? Maybe I'm missing something... but it feels to me that these types should automatically be adjusted.